Val dev merged reordered

4xdk opened the pull request:
Hopefully I didn't remove anything that was needed. 
This is related to issues https://github.com/PursuanceProject/pursuance/issues/122 and https://github.com/PursuanceProject/pursuance/issues/123

v-stickykeys committed:
test travis multi language

v-stickykeys committed:
test travis multi language

v-stickykeys committed:
attempt 2 for travis multi language

v-stickykeys committed:
attempt 2 for travis multi language

v-stickykeys committed:
attempt 3

v-stickykeys committed:
attempt 3

v-stickykeys committed:
filter all WIP

v-stickykeys committed:
filter all WIP

v-stickykeys committed:
Merge branch 'develop' of https://github.com/PursuanceProject/pursuance into develop

:x

v-stickykeys committed:
ordering public pursuances. wip

v-stickykeys committed:
ordering public pursuances. wip

v-stickykeys committed:
merge upstream to stay up to date

v-stickykeys committed:
order public pursuances

v-stickykeys committed:
order public pursuances

v-stickykeys committed:
refine public order reducer

v-stickykeys committed:
refine public order reducer

v-stickykeys committed:
merge upstream

v-stickykeys committed:
revert travis.yml and create dropdown for pursuances

v-stickykeys committed:
revert travis.yml and create dropdown for pursuances

v-stickykeys committed:
add newline char

v-stickykeys committed:
add newline char

v-stickykeys committed:
sort your pursuances a..z

v-stickykeys committed:
sort your pursuances a..z

v-stickykeys committed:
ignore pursuance-insert.txt'

v-stickykeys committed:
ignore pursuance-insert.txt'

v-stickykeys committed:
Merge https://github.com/PursuanceProject/pursuance into develop

stay up to date

v-stickykeys committed:
jump to pursuance from navbar

v-stickykeys committed:
jump to pursuance from navbar

v-stickykeys committed:
Merge branch 'develop' of https://github.com/PursuanceProject/pursuance into develop

stay up to date

v-stickykeys committed:
pursuance name in tasks and jump to dropdown

v-stickykeys committed:
pursuance name in tasks and jump to dropdown

v-stickykeys committed:
add radix to parseInt

v-stickykeys committed:
add radix to parseInt

v-stickykeys committed:
add about view for each pursuance

v-stickykeys committed:
add about view for each pursuance

v-stickykeys committed:
change id on taskhierarchy. use lowercase to compare

v-stickykeys committed:
change id on taskhierarchy. use lowercase to compare

v-stickykeys committed:
fix typos

v-stickykeys committed:
fix typos

v-stickykeys committed:
route and reduxify navbar

v-stickykeys committed:
route and reduxify navbar

v-stickykeys committed:
Merge branch 'develop' of https://github.com/thevaleriemack/pursuance into develop

stay up to date

v-stickykeys committed:
refreshing page shows correct pursuance in dropdown

v-stickykeys committed:
styling for about view and navbar

v-stickykeys committed:
Merge branch 'develop' into develop

elimisteve committed:
TaskHierarchy, JumpToPursuance switcher now updated without hard page refresh

elimisteve committed:
Whitespace

elimisteve committed:
Fixed merge conflicts

elimisteve committed:
PursuanceMenuItem: now re-renders correctly/displays active item in purple

Fixes #162

elimisteve committed:
Merge branch 'develop' into val-dev-merged-reordered

elimisteve committed:
PursuanceMenuItem: now highlights 'Tasks' button (as purple) when it should

elimisteve committed:
AboutView: tweaked styling, verbiage.  Whitespace.

elimisteve committed:
PublicPursuances: narrowed scope of CSS

Commit:
clear compile warnings

Commit:
fixing header html/css - End of day, this should be squashed later

Commit:
wip

Commit:
fixing header pursuance switches css, attempting to fix functionality

Commit:
fix to update properly instead of reloading page

Commit:
fix changing url/history after selection

Commit:
running linter

elimisteve commented on src/App.css:
```diff
@@ -26,3 +26,13 @@
+  -moz-user-select: none; /* Firefox */
+  -ms-user-select: none; /* Internet Explorer/Edge */
+  user-select: none; /* Non-prefixed version, currently supported by Chrome and Opera */
+}
```
@4xdk Do you see this used anywhere?

4xdk replied:
Ah nice spot. I've used it on the menu but it went missing during refactoring I'll add it back to nav-pursuances!

elimisteve commented on src/components/Content/Pursuance/PursuanceMenu.js:
```diff
@@ -18,6 +19,13 @@ const PursuanceMenu = ({ currentPursuanceId, location }) => {
+          action='about'
+          icon={<Info size={28} />}
+        />
+        <PursuanceMenuItem
```
@4xdk The latest version of this (in `develop`) is different; we no longer need to pass in `pursuanceId` and `location` explicitly, as `PursuanceMenuItem` is now reduxified.

elimisteve commented on src/components/Content/Pursuance/PursuanceMenuItem.js:
```diff
@@ -9,10 +9,15 @@ const PursuanceMenuItem = ({ pursuanceId, label, action, icon, location, classNa
+            || 
+            location.pathname.indexOf(`/${action}`) !== -1
+          ))
+        )
```
There is also more to this function now in `develop`; it treats the `discuss` action differently, which is what we want.

Commit:
PR-219 fixes

4xdk commented on src/components/PublicPursuances/PublicPursuances.css:
```diff
@@ -0,0 +1,72 @@
+  font-weight: normal;
+}
+
+.custom-dropdown {
```
This dropdown doesn't seem to match the style of other dropdowns, but still looks good IMO.

![screenshot_20180706_200128](https://user-images.githubusercontent.com/35820227/42395913-0d2b44c8-814f-11e8-9d3c-3f8fea03a131.png)

As an aside, I couldn't find a way to navigate to pursuances/all normally.

Commit:
small CR tweaks

Commit:
refactoring order -> sort, updating to match site dropdown style

Commit:
add a little bit more spacing

4xdk commented:
I've updated to match site dropdown style, also changed to say 'sort' instead.
![screenshot_20180707_213420](https://user-images.githubusercontent.com/35820227/42414464-597aaa2c-8225-11e8-9d18-b414710b97fd.png)

Commit:
Merge branch 'develop' into val-dev-merged-reordered

elimisteve commented on src/components/PublicPursuances/PublicPursuances.css:
```diff
@@ -1,72 +1,67 @@
 
-.custom-dropdown::before {
-  background-color: rgba(0,0,0,.15);
+#dashboard .sort .dropdown.open > .dropdown-menu {
```
@4xdk Do you like using the `>` for efficiency? It makes things more brittle, but I don't know about the speed differences.

4xdk replied:
TBH I just prefer to be reasonably specific to avoid styling something I didn't intend to. Especially working with unfamiliar code or overly generic classes as is the case here.

I wouldn't worry about CSS performance at all until we get to some serious animations.
